packages/babel-helper-simple-access/src/index.js

```diff
@@ -0,0 +1,87 @@
+import * as t from "babel-types";
+
+export default function simplifyAccess(path: NodePath, bindingNames) {
+  path.traverse(simpleAssignmentVisitor, {
+    scope: path.scope,
+    bindingNames,
+    seen: new WeakSet(),
+  });
+}
+
+const simpleAssignmentVisitor = {
+  UpdateExpression: {
+    exit(path) {
+      const { scope, bindingNames } = this;
+
+      const arg = path.get("argument");
+      if (!arg.isIdentifier()) return;
+      const localName = arg.node.name;
+
+      if (!bindingNames.has(localName)) return;
+
+      // redeclared in this scope
+      if (scope.getBinding(localName) !== path.scope.getBinding(localName)) {
+        return;
+      }
+
+      if (
+        path.node.prefix ||
+        (path.parentPath.isExpressionStatement() && !path.isCompletionRecord())
+      ) {
+        // ++i => (i += 1);
+        path.replaceWith(
+          t.assignmentExpression("+=", arg.node, t.numericLiteral(1)),
```

Comment:
Can we transform into a get and set? `super.prop` and `this.#foo` again.

Reply:
It's just translating into this to avoid code duplication. The `AssignmentExpression` visitor should still run on cases like this after this replacement happens, AFAIK, which will replace it with a get and a set.